data/src/main/java/com/linkedin/data/schema/annotation/SchemaAnnotationHandler.java

```diff
@@ -144,6 +149,105 @@ default SchemaVisitor getVisitor()
     return new PathSpecBasedSchemaAnnotationVisitor(this);
   }
 
+  /**
+   * Check annotations changes are compatible or not
+   *
+   * @param olderResolvedProperties the older resolvedProperties
+   * @param newerResolvedProperties the newer resolvedProperties
+   * @param olderContext the older annotationCheck context
+   * @param newerContext the newer annotationCheck context
+   * @return AnnotationCompatibilityResult
+   */
+  default AnnotationCompatibilityResult annotationCompatibilityCheck(Map<String,Object> olderResolvedProperties, Map<String, Object> newerResolvedProperties,
```

Comment:
Maybe we can have a new interface that specifically deals compatibility check.
My reason being the action to check compatibility maybe not needed for every schema annotation handler, so we create a new interface the inherit schema annotation handler, to add some more specific functions to schema annotation handler.

That way, you don't worry about using default here, and leave original schemaAnnotationHandler some flexibility.

Reply:
We had such discussion during the design phase. The main idea to put it into the SchemaAnnotationHandler is to complete the features of Schema Annotation Processor. In the handler, users provide how to resolve a custom annotation, how to validate a custom annotation. The compatibility check part was missing.